src/base/b-dynamic-page/b-dynamic-page.ts

```diff
@@ -477,14 +483,26 @@ export default class bDynamicPage extends iDynamicPage {
 				}
 
 				let
-					newPage = e;
+					newPageInfo = e;
 
 				if (Object.isTruly(this.eventConverter)) {
-					newPage = Array.concat([], this.eventConverter).reduce((res, fn) => fn.call(this, res, this.page), newPage);
+					newPageInfo = Array
+						.concat([], this.eventConverter)
+						.reduce((res, fn) => fn.call(this, res, this.page), newPageInfo);
 				}
 
-				if (newPage == null || Object.isString(newPage)) {
-					this.page = <string>newPage;
+				const
+					[newPageComponentName, newPageKey = newPageComponentName] = newPageInfo ?? [],
```

Comment:
Мб, просто `newPage` оставить как было?

```suggestion
					[newPage, newPageKey = newPage] = newPageInfo ?? [],
```

Reply:
Лично меня newPage путает сильно, потому что это может быть и названием роута, и названием компонента. Предпочитаю специфичность